Core/src/main/java/com/theprogrammingturkey/comz/game/Game.java

```diff
@@ -268,7 +254,20 @@ public Game(String name)
 	 */
 	public PlayerWeaponManager getPlayersWeapons(Player player)
 	{
-		return playersGuns.computeIfAbsent(player, PlayerWeaponManager::new);
+		for (Player pl : getPlayers()) {
+			if (pl.equals(player)) {
+				return gamePlayers.get(player).getWeaponManager();
+			}
+		}
+		return null;
+	}
```

Comment:
```suggestion
		Player pl = getPlayers().stream().filter(pl -> pl.equals(player)).findAny().orElse(null);
		if (pl == null) return pl;
		return gamePlayers.get(pl).getWeaponManager();
	}
```

**DO NOT APPLY BLINDLY.** This needs testing.

Reply:
That won't function, it wants a PlayerWeaponManager, not a player, so if it's null, it won't function

Reply:
Oh wait, that's if player is null, could be a little confusing